src/main/java/edu/wpi/grip/core/Step.java

```diff
@@ -67,7 +67,7 @@ public void onInputSocketChanged(SocketChangedEvent e) {
         final Socket socket = e.getSocket();
 
         // If this socket that changed is one of the inputs to this step, run the operation with the new value.
-        if (socket.getStep() == this && socket.getDirection().equals(Socket.Direction.INPUT)) {
+        if (socket.getStep().isPresent() && socket.getStep().get() == this && socket.getDirection().equals(Socket.Direction.INPUT)) {
```

Comment:
Could you clarify these comments?  In particular,
* `ifPresent` doesn't return anything, and I'm not sure why it would matter if it did.
* I don't think `Optional::equals` is ever going to change on us, and I don't think I've ever said I did.  We couldn't change it in our code if we wanted to, since `Optional` is `final`, but I'm not sure why that matters since we would never do that.

All I'm suggestion is that we make idiomatic use of the `java.util.Optional` class.